### ISSUE TITLE
Suggestion: Rename 2 methods: Headers -> EOH, Body -> EOM

### ISSUE THREAD

**Issue by iredmail** [emersion/go-milter#25]
Dear all,

After read the Sendmail Milter document, i think 2 interface methods are not clear enough and i suggest renaming them:

- `Headers()` -> `EOH()` (End of Headers)
- `Body()` -> `EOM()` (End of Message)

Both Sendmail and Postfix documents use "EOH" and "EOM", and their meanings are clearer.
If this suggestion is accepted, i will send a new PR to rename them.

**Comment by emersion** [comment]
These 3-letter codes are pretty cryptic TBH. Maybe `EndOfHeader`/`EndOfMessage` would be more appropriate...

**Comment by iredmail** [comment]
> These 3-letter codes are pretty cryptic TBH.

Agree. But since this library is all about milter, and "EOH", "EOM" are used by milter doc, it should be clear enough in our situation.

But i'm ok with EndOfHeader and EndOfMessage too. Just give me the final decision. :)
By the way, we should use `EndOfHeaders` (ends with a `s`) instead of `EndOfHeader`.

**Comment by emersion** [comment]
> By the way, we should use EndOfHeaders (ends with a s) instead of EndOfHeader.

The RFC folks say that it's an abuse to use "headers" (plural) when referring to a single message. Each message has exactly one _header_ and multiple _header fields_.

**Comment by tdewolff** [comment]
Agreed with renaming the methods, it was confusing when reading about milters in other places.

Another confusion is that you can't modify headers from the HeaderField (currently Header) nor EndOfHeader (currently Headers) functions, but they do supply a `Modifier` argument, why? Same for Connect, Helo, RcptTo, MailFrom, and BodyChunk functions.
